src/tree.cpp

```diff
@@ -0,0 +1,37 @@
+#include "tree.h"
+#include <stdlib.h>
+template <typename T>
+Tree<T>::Tree(int val, int parent_id){
+    this->dato = val;
+    this->parent_id = parent_id;
+    this->Left = NULL;
+    this->Right = NULL;
+}
+template <typename T>
+bool Tree<T>::Addchild(int val1, int val2) {
```

Comment:
`bool Tree<T>::Addchild(int val1=NULL, int val2=NULL) {`


What if you pass the Tree instead of create it?